my_enums.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+
+module Enumerable #:nodoc: all
+  def my_each
+    length.times do |n|
+      current = self[n]
+      yield(current)
+    end
+  end
+
+  def my_each_with_index
+    length.times do |n|
+      index = n
+      current = self[n]
+      yield(index, current)
+    end
+  end
+
+  def my_select
+    emp_arr = []
+    my_each do |n|
+      emp_arr << n if yield(n)
+    end
+    emp_arr
+  end
+
+  def my_all?
+    emp_arr = []
+    my_each do |n|
+      emp_arr << n if yield(n)
+    end
+    emp_arr == self
+  end
+
+  def my_any?
+    tester = false
+    my_each do |n|
+      tester = true if yield(n)
+      break if tester
+    end
+    tester
+  end
+
+  def my_none?
+    tester = true
+    my_each do |n|
+      tester = false if yield(n)
+      break if tester == false
+    end
+    tester
+  end
+
+  def my_count(item = 'CANTOR')
+    count = 0
+    if item == 'CANTOR'
+      count = length
+    else
+      my_each do |n|
+        count += 1 if n == item
+      end
+    end
+    count
+  end
+
+  def my_map_one
+    emp_arr = []
+    my_each do |n|
+      emp_arr << yield(n)
+    end
+    emp_arr
+  end
+
+  def my_map_two(&block)
+    emp_arr = []
+    my_each do |n|
+      emp_arr << block.call(n)
+    end
+    emp_arr
+  end
+
+  def my_map_three(param = nil)
+    emp_arr = []
+    my_each do |n|
+      if param.nil? && block_given?
+        emp_arr << yield(n)
+      elsif !param.nil? && block_given?
+        emp_arr << param.call(n)
+      end
+    end
+    emp_arr
+  end
+
+  def my_inject(inivalue = nil, symbol = nil)
+    if !inivalue.nil? && !symbol.nil?
+      my_each { |num| inivalue.method(symbol).call(num) }
+      inivalue
+    elsif !inivalue.nil? && inivalue.is_a?(Symbol) && symbol.nil?
+      memo, *rem_elements = self
+      rem_elements.my_each { |num| memo = memo.method(inivalue).call(num) }
+      memo
+    elsif !inivalue.nil? && inivalue.is_a?(Integer) && symbol.nil?
+      my_each { |num| inivalue = yield(inivalue, num) }
+      inivalue
+    elsif inivalue.nil? && symbol.nil?
+      inivalue, *rem_elements = self
+      rem_elements.my_each { |num| inivalue = yield(inivalue, num) }
+      inivalue
+    end
+  end
+
```

Comment:
- [ ] enumerables my_inject combines all elements of enum by applying a binary operation, specified by a block
- [ ] enumerables my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method